FSAE.X/FSAE_spi.c

```diff
@@ -0,0 +1,336 @@
+#include "FSAE_spi.h"
+
+void init_spi1(int mhz, int size) {
+  unlock_config();
+
+  // Initialize SDI1/SDO1 PPS pins
+  CFGCONbits.IOLOCK = 0;
+  TRISBbits.TRISB9 = INPUT;
+  SDI1Rbits.SDI1R = 0b0101; // RPB9
+  TRISBbits.TRISB10 = OUTPUT;
+  RPB10Rbits.RPB10R = 0b0101; // SDO1
+  CFGCONbits.IOLOCK = 1;
+
+  // Initialize SCK1
+  TRISDbits.TRISD1 = OUTPUT;
+
+  // Disable interrupts
+  IEC3bits.SPI1EIE = 0;
+  IEC3bits.SPI1RXIE = 0;
+  IEC3bits.SPI1TXIE = 0;
+
+  // Disable SPI1 module
+  SPI1CONbits.ON = 0;
+
+  // Clear receive buffer
+  uint32_t readVal = SPI1BUF;
+
+  // Use standard buffer mode
+  SPI1CONbits.ENHBUF = 0;
+
+  /**
+   * F_SCK = F_PBCLK2 / (2 * (SPI1BRG + 1))
+   * SPI1BRG = (F_PBCLK2/(2*F_SCK))-1
+   */
+
+  // Set the baud rate (see above equation, ceil will round to lower freq)
+  SPI1BRG = ceil((PBCLK2/(2.0*mhz))-1);
```

Comment:
I think it might be better to have standard define statements for the BRG values for these. Like `SPI_1MHZ`, `SPI_500KHZ`, `SPI_20MHZ`, etc. Then we can add non-clean number frequencies if need be. I've definitely had to do frequencies below 1Mhz before.